packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -77,7 +77,11 @@ export default class OptionManager {
     let obj: ?PluginObject;
 
     if (typeof fn === "function") {
-      obj = fn(context);
+      try {
+        obj = fn(context);
+      } catch (e) {
+        throw new Error(`You are probably running a Babel 5 plugin (${alias}) with Babel 6. Check http://babeljs.io/docs/plugins/ for official plugins or search npm.`);
```

Comment:
We should check `e.message` I think, otherwise we risk this making plugin development a pain, since this will swallow errors.